hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileWriter.java

```diff
@@ -104,9 +106,13 @@ public HoodieHFileWriter(String instantTime, Path file, HoodieHFileConfig hfileC
 
   @Override
   public void writeAvroWithMetadata(R avroRecord, HoodieRecord record) throws IOException {
-    prepRecordWithMetadata(avroRecord, record, instantTime,
-        taskContextSupplier.getPartitionIdSupplier().get(), recordIndex, file.getName());
-    writeAvro(record.getRecordKey(), (IndexedRecord) avroRecord);
+    if (populateMetaFields) {
```

Comment:
Is this change backward compatible. On disk format of metadata records are now going to miss out on the meta fields. After the downgrade would this break anything?